main/zoe/guide/contracts/autoswap.md

```diff
@@ -2,161 +2,170 @@
 
-Create an instance of the Autoswap code, and receive an invite that
-when used, will let you add liquidity to Autoswap.
+When someone creates an instance of the Autoswap code, they receive only the public
+facet. The creator has no special access or priveleges.
 
 ```js
 const issuerKeywordRecord = harden({
-  TokenA: moolaIssuer,
-  TokenB: simoleanIssuer,
+  Central: moolaIssuer,
+  Secondary: simoleanIssuer,
 });
 
-const addLiquidityInvite = await E(zoe).makeInstance(
-  autoswapInstallationHandle,
+const publicFacet = await E(zoe).startInstance(
+  autoswapInstallation,
   issuerKeywordRecord,
 );
 ```
 
 ## Adding liquidity to the pool
 
-We can contribute to the Autoswap liquidity pool by making an offer by
-using an addLiquidityInvite. For instance, let's say that Alice
-creates a proposal with the associated payments of moola and simoleans
-and escrows them:
+We can contribute to the Autoswap liquidity pool by making an offer using an
+addLiquidityInvitation. For instance, let's say that Alice creates a proposal with the
+associated payments of moola and simoleans and escrows them. At the time she's creating
+the pool, the market price is 2 moola for 1 simolean, so that's the rate she sets up.
 
 ```js
-// Alice adds liquidity
-// 10 moola = 5 simoleans at the time of the liquidity adding
-// aka 2 moola = 1 simolean
-
 const moola = moolaAmountMath.make;
 const simoleans = simoleanAmountMath.make;
 const liquidity = liquidityAmountMath.make;
+const liquidityIssuer = await E(zoe).getLiquidityIssuer();
+const aliceLiquidityPurse = E(liquidityIssuer).makeEmptyPurse();
 
 const aliceProposal = harden({
   give: {
-    TokenA: moola(10),
-    TokenB: simolean(5)
+    Central: moola(10),
+    Secondary: simolean(5)
   },
   want: { Liquidity: liquidity(10) },
   exit: { onDemand: null },
 ]);
 
 const alicePayments = {
-  TokenA: aliceMoolaPayment,
-  TokenB: aliceSimoleanPayment
+  Central: aliceMoolaPayment,
+  Secondary: aliceSimoleanPayment
 }
 
-const {
-  outcome,
-  payout: aliceAddLiquidityPayoutP,
-} = await E(zoe).offer(addLiquidityInvite, aliceProposal, alicePayments);
+const seat =
+  await E(zoe).offer(addLiquidityInvitation, aliceProposal, alicePayments);
+const liquidityPayment = await E(seat).getPayout('Liquidity');
 
+E(aliceLiquidityPurse).deposit(liquidityPayment);
 ```
-She is able to ensure that she will get a minimum number of liquidity
-tokens back by specifying a rule for the liquidity token slot with
-`want`. In this case, Alice is stating that she wants at least
-10 liquidity tokens back.
 
 ## Making a swap offer
 
-Let's say that Bob wants to use the moola<->simolean Autoswap
-to exchange 2 moola. First he will check the price using the public
-API:
-
-#### Public API:
-1. `getCurrentPrice`
-2. `getLiquidityIssuer`
-3. `getPoolAllocation`
-4. `makeInvite`
+Let's say that Bob wants to use the mool-to-simolean Autoswap to sell 2 moola for
+simolean. First he checks the price using the publicFacet:
```

Comment:
"First, he uses the `publicFacet` to check the price."

Reply:
done